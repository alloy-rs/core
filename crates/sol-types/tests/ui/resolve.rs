use alloy_sol_types::sol;

sol! {
    struct A {
        B a;
    }

    struct B {
        A a;
    }
}

sol! {
    struct A {
        B a;
    }

    struct B {
        C c;
    }

    struct C {
        A a;
    }
}

sol! {
    struct A {
        B a;
    }

    struct B {
        C c;
    }

    struct C {
        D d;
    }

    struct D {
        A a;
    }
}

fn main() {}
