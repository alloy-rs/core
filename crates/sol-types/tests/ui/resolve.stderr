error: failed to resolve types.
       This is likely due to an infinitely recursive type definition.
       If you believe this is a bug, please file an issue at https://github.com/alloy-rs/core/issues/new/choose
 --> tests/ui/resolve.rs:4:12
  |
4 |     struct A {
  |            ^

error: failed to resolve types.
       This is likely due to an infinitely recursive type definition.
       If you believe this is a bug, please file an issue at https://github.com/alloy-rs/core/issues/new/choose
  --> tests/ui/resolve.rs:14:12
   |
14 |     struct A {
   |            ^

error: failed to resolve types.
       This is likely due to an infinitely recursive type definition.
       If you believe this is a bug, please file an issue at https://github.com/alloy-rs/core/issues/new/choose
  --> tests/ui/resolve.rs:28:12
   |
28 |     struct A {
   |            ^
